project-koku/koku
# Adding infrastructure field to provider response

Addresses #607 

The provider endpoint now has a new field called `infrastructure` which will designate what infrastructure a provider is running on.

The primary use case for the short term is OCP on AWS.  OCP clusters running on AWS (resource_id matches and usage period aligns between the two providers) will now have `infrastructure=AWS` when we detect that an OCP cluster is running on AWS.

All providers not detected to be running on AWS have `infrastructure=Unknown`.  If we encounter an error while trying to determine the infrastructure the API response has `infrastructure=Unknown-Error`.

There is still room for improvement to cleanup the test data generation.  First candidate would be to decouple the `FakeAWSCostData` from the `ReportQueryTest` test suite and then update the new `OCPAWSReportDataGenerator().add_aws_data_to_tenant()` to use it instead.  I started doing this work but I was hitting the point of diminishing returns for focusing on new sprint content.

**Testing**
1.  Add AWS provider with data and ingest
2. Add OCP provider with matching date range and resource_id
3. Check the provider list and see that both AWS and OCP are infrastructure="Unknown"
4.  Ingest the OCP data and recheck the infrastruture field.  Verify that it is "AWS"
5.  Add another OCP Provider and verify that the Insfrastructure is "Unknown" and the original OCP-on-AWS provider is "AWS"

**Test Results**
[koku_607_ut.txt](https://github.com/project-koku/koku/files/2988471/koku_607_ut.txt)
